testapi.pm

```diff
@@ -475,6 +491,43 @@ sub wait_screen_change(&@) {
 
 ## helpers end
 
+## synchronization API
+use constant {
+    MUTEX_CREATE => 'mutex_create',
+    MUTEX_LOCK   => 'mutex_lock',
+    MUTEX_UNLOCK => 'mutex_unlock',
+};
+
+sub mutex_lock($) {
+    my ($name) = @_;
+    return _mutex_call(MUTEX_LOCK, $name);
+}
+
+sub mutex_unlock($) {
+    my ($name) = @_;
+    return _mutex_call(MUTEX_UNLOCK, $name);
+}
+
+sub mutex_create($) {
+    my ($name) = @_;
+    return _mutex_call(MUTEX_CREATE, $name);
+}
+
+sub _mutex_call($$) {
+    my ($action, $name) = @_;
+    # blocking call to worker, worker asks Scheduler for lock
+    bmwqemu::fctlog("Trying mutex action $action on lock $name");
+    my $port = $bmwqemu::vars{'WORKER_PORT'};
+    my $path = "$action/$name";
+    while (1) {
+        my $res = qx(curl -s -o /dev/null -w \"\%{http_code}\" -I http://localhost:$port/$path);
```

Comment:
and there is really no way you can make use of the HTTP interface already available in os-autoinst? like a long poll from the worker to it?

I find this whole interaction model very complex and hard to explain ;(